### ISSUE TITLE
Play Button delay - Android device 

### ISSUE THREAD

**Issue by wkande** [24hourkirtan/audio-streaming-app#2]
There is a 20 second delay until music plays after selecting the Play button.  Music does stop right away when the Pause button is selected.

Deployed to Samsung S4 SCH-1545
Android 5.0.1

I tried building with SDKs 22, 23, and 24 al behave the same.

**Comment by schmoofer** [comment]
@wkande How long is the delay when pressing play at www.24hourkirtan.fm? I'm currently experiencing only a max 5 second delay on my LG V10, so it seems it is either hardware-dependent, network-dependent, or a combination of both. @nandababa What delays are you experiencing and what phone are you using? To be honest, this isn't something I can directly address since any sort of further optimization, if possible, would involve native code optimization which wouldn't be trivial and cost-efficient.

**Comment by wkande** [comment]
Using my Mac Mini with Chrome I consistently get around 4.5 seconds. Again this morning the Samsung S4 is running at 21 seconds. The Mac Mini is wired ethernet and the Samsung is WIFI. I also tried my Mac AirBook using the same WFI that the Samsung S4 is on and it starts within 5 seconds as well. Not sure if the app attempting to buffer before playing on the Samsung S4. After I build an Android alpha version to put up on the WIKI this morning I will do a download from there to see if the this persists.

**Comment by nandababa** [comment]
I was able to install the APK @wkande sent me on my HTC One. When I click the start listening button it consistently takes 8-9 seconds on my home WIFI. When I listen to 24 Hour Kirtan Radio via TuneIn on the HTC One it takes 4 seconds for buffering and starting to play.

I also noticed that the "Now Playing" info does not show up as in the iOS simulator and the Dance Script font does is not being applied to the heading, either. If @wkande confirms that the APK is based on the latest code from the repository, I will create separate issues for that.

**Comment by wkande** [comment]
The test build (release) APK from yesterday was from around noon. I saw that it contained the code change to icecast so it should have been the latest. I will try to do an official alpha release today after pulling down a repository update.

**Comment by nandababa** [comment]
Thanks for confirming. Does your Samsung S4 show the "Now Playing" info?

**Comment by wkande** [comment]
It does not appear as so. As you can see from the screen shot below the music is playing and the pause button is visible. The text says "Start Listening" even though the music is playing.

![screenshot_2016-02-08-04-31-05](https://cloud.githubusercontent.com/assets/3522283/12884678/243c7674-ce1d-11e5-9d79-a095977be4d1.png)

**Comment by wkande** [comment]
Here is my latest test using the same Wifi access point for all 4 test below.

1. Mac AirBook > Safari > http://24hourkirtan.fm/ > 4 seconds
2. Samsung > native browser > http://24hourkirtan.fm/ > 21 seconds
3. Samsung > Test APK > 21 seconds
4. Samsung > **crosswalk browser** > APK > 21 seconds

You will notice the fourth test I build the project using the crosswalk browser thinking this might have an impact but no luck.

------------------------------------------------------
Console output from **android-debug.APK**
The first two lines output during app startup, the rest outputted while waiting the 21 seconds for music to play.
-------------------------------------------------------

```
No Content-Security-Policy meta  tag found. Please add one when using the cordova-plugin-whitelist plugin.(anonymous function) @ plugins/cordova-plugin-whitelist/whitelist.js:24
7whitelist.js:26 

No Content-Security-Policy meta tag found. Please add one when using the cordova-plugin-whitelist plugin.
index.html:1 

Uncaught (in promise) module org.apache.cordova.media.Media not found
services.js:17 

Object {data: Object, status: 200, config: Object, statusText: "OK"}
services.js:17 

Object {data: Object, status: 200, config: Object, statusText: "OK"}
whitelist.js:26 

No Content-Security-Policy meta tag found. Please add one when using the cordova-plugin-whitelist plugin.
services.js:17 

Object {data: Object, status: 200, config: Object, statusText: "OK"}
services.js:17

 Object {data: Object, status: 200, config: Object, statusText: "OK"}
whitelist.js:26 No Content-Security-Policy meta tag found. Please add one when using the cordova-plugin-whitelist plugin.
services.js:17 

Object {data: Object, status: 200, config: Object, statusText: "OK"}
cordova.js:59 

Uncaught module org.apache.cordova.media.Media not foundrequire @ cordova.js:59localRequire @ cordova.js:49(anonymous function) @ VM66:1processMessage @ cordova.js:1070processMessages @ cordova.js:1104pollOnce @ cordova.js:973pollOnceFromOnlineEvent @ cordova.js:960
index.html:1 

Uncaught (in promise) module org.apache.cordova.media.Media not found
index.html:1 

Uncaught (in promise) module org.apache.cordova.media.Media not found
services.js:17 

Object {data: Object, status: 200, config: Object, statusText: "OK"}
whitelist.js:26 

No Content-Security-Policy meta tag found. Please add one when using the cordova-plugin-whitelist plugin.
index.html:1 

Uncaught (in promise) module org.apache.cordova.media.Media not found
whitelist.js:26 

No Content-Security-Policy meta tag found. Please add one when using the cordova-plugin-whitelist plugin.
```

**Comment by schmoofer** [comment]
Based on those results, the delay in initializing the audio stream seems to be solely hardware-dependent. Not much I can do there.

**Comment by DavidFrahm** [comment]
All those `Media not found` seem like something that could be fixed in the app code.  Maybe the lack of a proper CSP tag is tripping it up for awhile, then finally allowing the http request through?

(just brainstorming here; interesting problem)

**Comment by schmoofer** [comment]
@DavidFrahm the ```Media not found``` problem is due to the audio streaming plugin having its own version of the media plugin that excludes the Media object.

**Comment by schmoofer** [comment]
Partially addressed in ea6a6510a4c95799e026202386db7c1c06ce305a
